crates/web-plugins/didcomm-messaging/src/did_rotation/did_rotation.rs

```diff
@@ -0,0 +1,350 @@
+use super::errors::RotationError;
+use crate::{didcomm::bridge::LocalDIDResolver, model::stateful::entity::Connection};
+use axum::response::{IntoResponse, Response};
+use database::Repository;
+use didcomm::{FromPrior, Message};
+use mongodb::bson::doc;
+use std::sync::Arc;
+
+/// https://identity.foundation/didcomm-messaging/spec/#did-rotation
+pub async fn did_rotation(
+    msg: Message,
+    connection_repos: &Arc<dyn Repository<Connection>>,
+) -> Result<(), Response> {
+    // Check if from_prior is none
+    if msg.from_prior.is_none() {
+    } else {
+        let jwt = msg.from_prior.unwrap();
+        let did_resolver = LocalDIDResolver::default();
+
+        // decode and validate jwt signature
+        let (from_prior, _kid) = FromPrior::unpack(&jwt, &did_resolver)
+            .await
+            .map_err(|_| RotationError::InvalidFromPrior.json().into_response())?;
+        let prev = from_prior.iss;
+
+        // validate if did is  known
+        let _ = match connection_repos
+            .find_one_by(doc! {"client_did": &prev})
+            .await
+            .unwrap()
+        {
+            Some(mut connection) => {
+                // get new did for communication, if empty then we end the relationship
+                let new = from_prior.sub;
+
+                if new.is_empty() {
+                    let id = connection.id.unwrap_or_default();
+                    // end the relationship with the mediator
+                    connection_repos
+                        .delete_one(id)
+                        .await
+                        .map_err(|_| RotationError::TargetNotFound.json())
+                        .into_response();
```

Comment:
Shouldn't you return a result here?
```suggestion
                    return connection_repos
                        .delete_one(id)
                        .await
                        .map_err(|_| RotationError::TargetNotFound.json().into_response());
```